stanford_soe_r25.pages.inc

```diff
@@ -0,0 +1,994 @@
+      }
+      $form['markup2'] = array(
+        '#markup' => t($msg_text),
+      );
+      if ($op == 'cancel') {
+        $form['series'] = array(
+          '#type' => 'radios',
+          '#default_value' => 1,
+          '#options' => array(
+            1 => 'Cancel this occurrence',
+            2 => 'Cancel entire series'
+          ),
+        );
+      }
+      else {
+        $form['series'] = array(
+          '#type' => 'hidden',
+          '#value' => 2,
+        );
+      }
+    }
+    else {
+      $form['series'] = array(
+        '#type' => 'hidden',
+        '#value' => 2,
+      );
+    }
+  }
+  else {
+    // if we are back from "are you sure" and want to proceed with the operation
+    // then take the XML string and update the appropriate sections for occurence
+    // or entire event with the new event state
+    $xml_string = $result['raw-xml'];
+    $find_str = 'status="est"';
+    if ($form_state['input']['series'] == 1) {
+      $reservation_id = 0;
+      if (!empty($result['index']['R25:RESERVATION_START_DT']) &&
+        is_array($result['index']['R25:RESERVATION_START_DT'])
+      ) {
+        foreach ($result['index']['R25:RESERVATION_START_DT'] as $key => $value) {
+          if (!empty($result['vals'][$value]['value']) &&
+            $start == $result['vals'][$value]['value']
+          ) {
+            $reskey = $result['index']['R25:RESERVATION_ID'][$key];
+            $reservation_id = $result['vals'][$reskey]['value'];
+            break;
+          }
+        }
+        if ($reservation_id > 0) {
+          $resptr1 = strpos($xml_string, '<r25:reservation_id>' . $reservation_id);
+          if ($resptr1 !== FALSE) {
+            $pos = strpos($xml_string, $find_str);
+            if ($pos !== FALSE) {
+              $xml_string = substr_replace($xml_string, 'status="mod"', $pos, strlen($find_str));
+            }
+            $proptr = strpos($xml_string, '<r25:profile');
+            if ($proptr !== FALSE) {
+              $pos = strpos($xml_string, $find_str, $proptr);
+              if ($pos !== FALSE) {
+                $xml_string = substr_replace($xml_string, 'status="mod"', $pos, strlen($find_str));
+              }
+            }
+            $temp1 = substr($xml_string, 0, $resptr1);
+            $resptr2 = strrpos($temp1, $find_str);
+            if ($resptr2 !== FALSE) {
+              $xml_string = substr($xml_string, 0, $resptr2) . 'status="mod"' . substr($xml_string, $resptr2 + 12);
+            }
+            $resptr2 = strpos($xml_string, '<r25:reservation_state>', $resptr1);
+            $resptr3 = strpos($xml_string, '</r25:reservation_state>', $resptr1);
+            if ($resptr2 !== FALSE && $resptr3 !== FALSE) {
+              $xml_string = substr_replace($xml_string, '<r25:reservation_state>99</r25:reservation_state>', $resptr2, $resptr3 + 24 - $resptr2);
+            }
+          }
+        }
+      }
+    }
+    else {
+      if ($form_state['input']['series'] == 2) {
```

Comment:
form values should be used instead of form input.

Reply:
there are a number of these.